tools/tuf/test/Orbit-auto-update-test-guide.md

```diff
@@ -0,0 +1,151 @@
+
+```sh
+git checkout orbit-v1.2.0
+rm -rf test_tuf
+
+SYSTEMS="macos windows linux" \
+PKG_FLEET_URL=https://localhost:8080 \
+PKG_TUF_URL=http://localhost:8081 \
+DEB_FLEET_URL=https://host.docker.internal:8080 \
+DEB_TUF_URL=http://host.docker.internal:8081 \
+RPM_FLEET_URL=https://host.docker.internal:8080 \
+RPM_TUF_URL=http://host.docker.internal:8081 \
+MSI_FLEET_URL=https://host.docker.internal:8080 \
+MSI_TUF_URL=http://host.docker.internal:8081 \
+GENERATE_PKG=1 \
+GENERATE_DEB=1 \
+GENERATE_RPM=1 \
+GENERATE_MSI=1 \
+ENROLL_SECRET=$ENROLL_SECRET \
+FLEET_DESKTOP=1 \
+FLEET_CERTIFICATE=1 \
+./tools/tuf/test/main.sh
+```
+
+2. Install Orbit on the host (macOS):
+```sh
+sudo installer -pkg fleet-osquery.pkg -verbose -target /
+```
+
+3. Copy the generated packages into the VMWare shared folders:
+```sh
+cp fleet-osquery.msi ~/shared-windows
+cp fleet-osquery_42.0.0_amd64.deb ~/shared-ubuntu
+```
+
+4. Proceed to install Orbit in both VM hosts.
+
+5. Verify three hosts have enrolled (by running `./build/fleetctl get hosts` or using the browser).
+
+6. Verify the three Fleet Desktop instances are working, by clicking the "My device" menu item on the three OSs.
+
+## New Orbit release
+
+1. Now let's "release" new Orbit + Fleet Desktop version (via auto-update) by using latest `main`.
+
+```sh
+git checkout main
+
+## Windows
+
+# Compile a new version of Orbit for Windows:
+GOOS=windows GOARCH=amd64 go build -o orbit-windows.exe ./orbit/cmd/orbit
+# Push the compiled Orbit as a new version
+./tools/tuf/test/push_target.sh windows orbit orbit-windows.exe 43
+
+# Compile a new version of fleet-desktop for Windows:
+FLEET_DESKTOP_VERSION=43.0.0 make desktop-windows
+# Push the desktop target as a new version
+./tools/tuf/test/push_target.sh windows desktop fleet-desktop.exe 43
+
+## Linux
+
+# Compile a new version of Orbit for Linux:
+GOOS=linux GOARCH=amd64 go build -o orbit-linux ./orbit/cmd/orbit
+# Push the compiled Orbit as a new version
+./tools/tuf/test/push_target.sh linux orbit orbit-linux 43
+
+# Compile a new version of fleet-desktop for Linux:
+FLEET_DESKTOP_VERSION=43.0.0 make desktop-linux
+# Push the desktop target as a new version
+./tools/tuf/test/push_target.sh linux desktop desktop.tar.gz 43
+
+## macOS
+
+# Compile a new version of Orbit for macOS:
+GOOS=darwin GOARCH=amd64 go build -o orbit-darwin ./orbit/cmd/orbit
+# Push the compiled Orbit as a new version
+./tools/tuf/test/push_target.sh macos orbit orbit-darwin 43
+
+# Compile a new version of fleet-desktop for macOS:
+FLEET_DESKTOP_VERSION=43.0.0 make desktop-app-tar-gz
+# Push the desktop target as a new version
+./tools/tuf/test/push_target.sh macos desktop desktop.app.tar.gz 43
+```
+
+2. Wait for ~1m for all hosts to auto-update.
+
+3. Verify all hosts now show "Fleet Desktop v43.0.0" on the Fleet Desktop menu.
+
+4. Verify the three Fleet Desktop instances are working, by visiting "My device"
```

Comment:
I'd add "hit refresh in My Device" or run a live query, to make sure that osquery is still functioning properly and we're ingesting the data